ext/standard/string.c

```diff
@@ -1192,6 +1192,105 @@ PHP_FUNCTION(strtolower)
 }
 /* }}} */
 
+PHP_FUNCTION(str_increment)
+{
+	zend_string *str;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(str)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (ZSTR_LEN(str) == 0) {
+		zend_argument_value_error(1, "cannot be empty");
+		RETURN_THROWS();
+	}
+	if (!zend_string_only_has_ascii_alphanumeric(str)) {
+		zend_argument_value_error(1, "must be composed only of alphanumeric ASCII characters");
+		RETURN_THROWS();
+	}
+
+	zend_string *incremented = zend_string_init(ZSTR_VAL(str), ZSTR_LEN(str), /* persistant */ false);
+	size_t position = ZSTR_LEN(str)-1;
+	bool carry = false;
+
+	do {
+		char c = ZSTR_VAL(incremented)[position];
+		if (EXPECTED( (c >= 'a' && c < 'z') || (c >= 'A' && c < 'Z') || (c >= '0' && c < '9') )) {
+			carry = false;
+			ZSTR_VAL(incremented)[position]++;
+		} else { /* if 'z', 'Z', or '9' */
+			carry = true;
+			if (c == '9') {
+				ZSTR_VAL(incremented)[position] = '0';
+			} else {
+				ZSTR_VAL(incremented)[position] -= 25;
+			}
+		}
+	} while (carry && position-- > 0);
+
+	if (UNEXPECTED(carry)) {
+		zend_string *tmp = zend_string_alloc(ZSTR_LEN(incremented)+1, 0);
+		memcpy(ZSTR_VAL(tmp) + 1, ZSTR_VAL(incremented), ZSTR_LEN(incremented));
+		ZSTR_VAL(tmp)[ZSTR_LEN(incremented)+1] = '\0';
+		switch (ZSTR_VAL(incremented)[0]) {
+			case '0':
+				ZSTR_VAL(tmp)[0] = '1';
+				break;
+			default:
+				ZSTR_VAL(tmp)[0] = ZSTR_VAL(incremented)[0];
+				break;
+		}
+		zend_string_release_ex(incremented, /* persistant */ false);
+		RETURN_STR(tmp);
+	}
+	RETURN_STR(incremented);
+}
+
+
+PHP_FUNCTION(str_decrement)
+{
+	zend_string *str;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(str)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (ZSTR_LEN(str) == 0) {
+		zend_argument_value_error(1, "cannot be empty");
+		RETURN_THROWS();
+	}
+	if (!zend_string_only_has_ascii_alphanumeric(str)) {
+		zend_argument_value_error(1, "must be composed only of alphanumeric ASCII characters");
+		RETURN_THROWS();
+	}
+
+	zend_string *decremented = zend_string_init(ZSTR_VAL(str), ZSTR_LEN(str), /* persistant */ false);
+	size_t position = ZSTR_LEN(str)-1;
+	bool carry = false;
+
+	do {
+		char c = ZSTR_VAL(decremented)[position];
+		if (EXPECTED( (c > 'a' && c <= 'z') || (c > 'A' && c <= 'Z') || (c > '0' && c <= '9') )) {
+			carry = false;
+			ZSTR_VAL(decremented)[position]--;
+		} else { /* if 'a', 'A', or '0' */
+			carry = true;
+			if (c == '0') {
+				ZSTR_VAL(decremented)[position] = '9';
+			} else {
+				ZSTR_VAL(decremented)[position] += 25;
+			}
+		}
+	} while (carry && position-- > 0);
+
+	if (UNEXPECTED(carry)) {
+		zend_string_release_ex(decremented, /* persistant */ false);
```

Comment:
This should be almost equivalent to the logic above, I'd prefer if we could just implement this.